### PR TITLE
Add localized names to card editor

### LINKED ISSUE #3658: Add card name to Lovelace "Card Configuration" headings
**Home Assistant release with the issue:**
0.98.4

**Description of problem:**
The GUI editor for each and every Lovelace card currently has the same exact title/heading of "Card Configuration."  

This can make it easy for users to lose their place, and forget which card they are on. Especially for newer users or demo users who are browsing through all 24 cards in quick succession. These headings should display the name of the card: "Markdown Card Configuration" , "Sensor Card Configuration" and so forth, so that there is never any confusion.

Before and after:

image
image

### PR DESCRIPTION
**Pull request by timmo001**
### Description

This adds localized names to the card configuration, helping the user to see what card type they are editing.

I have also replaced the hardcoded names in the card picker to localized names. 

### Related Issue

Closes #3658

### Screenshots

image

image

image

---

This populates the heading as soon as it is typed/recognized in the json editor also:

image

image

### CODE REVIEW

File: src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts [hunk 1]
```diff
@@ -62,10 +62,19 @@ export class HuiDialogEditCard extends LitElement {
       return html``;
     }
 
+    let heading: string = this.hass!.localize(
+      "ui.panel.lovelace.editor.edit_card.header"
+    );
```
Comment (bramkragten) [inline src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts]: Do this in the `else`

File: src/translations/en.json [hunk 2]
```diff
@@ -1375,17 +1375,31 @@
             "migrate": "Migrate config"
           },
           "card": {
+            "alarm-panel": {
+              "name": "Alarm panel"
+            },
             "alarm_panel": {
```
Comment (bramkragten) [inline src/translations/en.json]: Duplicate `alarm-panel` and `alarm_panel`?

File: src/panels/lovelace/editor/card-editor/hui-card-picker.ts [hunk 3]
```diff
@@ -11,49 +11,154 @@ import "@material/mwc-button";
...
-];
-
 @customElement("hui-card-picker")
 export class HuiCardPicker extends LitElement {
   public hass?: HomeAssistant;
 
   public cardPicked?: (cardConf: LovelaceCardConfig) => void;
 
   protected render(): TemplateResult | void {
+    const cards: CardOption[] = [
```
Comment (bramkragten) [inline src/panels/lovelace/editor/card-editor/hui-card-picker.ts]: Why was this moved to `render`? This is not needed to be recalculated every render.
You can move it to updated and update it only if the language is changed.

Comment (bramkragten) [reply src/panels/lovelace/editor/card-editor/hui-card-picker.ts]: Better yet, we don't need the name here, we can just use this in the render:
```js
this.hass!.localize(`ui.panel.lovelace.editor.card.${card.type}.name`)
```

File: src/translations/en.json (hunk 2 above)
Comment (timmo001) [reply src/translations/en.json]: There's an existing entry for `alarm_panel` which doesn't conform to the rest of the 'standard' in that json obj.

I could move the items and remove that one but would have to do a find and replace in the repo which will increase the risk somewhat.

Up to you guys what we should do 👍

Comment (bramkragten) [reply src/translations/en.json]: Let's rename it

Comment (timmo001) [reply src/translations/en.json]: Done

File: src/panels/lovelace/editor/card-editor/hui-card-picker.ts (hunk 3 above)
Comment (timmo001) [reply src/panels/lovelace/editor/card-editor/hui-card-picker.ts]: Decided to remove the object entirely in favor for a string array. and use your above suggestion

**Review by bramkragten — APPROVED** [review]
Thanks! 🦁
